### PR TITLE
Fix opening partition that is marked as deleted

### LINKED ISSUE #59: Recreating a deleted partition may be undefined behaviour
Reproduce:
1. Delete a partition
2. A partition may not be cleaned up until all handles to it are dropped, the partition folder is marked with a `.deleted` marker
3. At this point, the keyspace forgot about the partition, now recreate the partition
4. No idea what happens: Need to check if the partition folder exists AND `.deleted` marker exists
5. When the handles of the "old" partition are dropped, the partition folder will be deleted, even after we "recreated" it, see step 4 for solution

### PR DESCRIPTION
**Pull request by dharanad**
## Which issue does this PR close?
Closes #59 

## Are these changes tested?
Tested changes maually as suggested in #59

### CODE REVIEW

File: src/keyspace.rs [hunk 1]
```diff
@@ -372,7 +372,13 @@ impl Keyspace {
         let mut partitions = self.partitions.write().expect("lock is poisoned");
 
         Ok(if let Some(partition) = partitions.get(name) {
-            partition.clone()
+            let handle = partition.clone();
+            /// throw error if [PARTITION_DELETED_MARKER] exists with a existing partition
+            if handle.path().join(PARTITION_DELETED_MARKER).exists() {
+                log::error!("Failed to open partition, partition is deleted.");
+                return Err(Error::PartitionDeleted);
+            }
```
Comment (marvin-j97) [inline src/keyspace.rs:380]: This should be in the `else` case.

When the partition is deleted, it is not part of `partitions` anymore. I guess there could be a race condition, where `Keyspace::delete_partition` is not finished yet, but that could be fixed by wrapping the entire function in the write lock.

Anyway, if you create a new test file, do something like this:

```rs
let keyspace = ...;
let partition = keyspace.open_partition("default", Default::default())?;
keyspace.delete_partition(partition.clone())?;

// This should throw the PartitionDeleted error variant...
assert!(keyspace.open_partition("default", Default::default).is_err());

// Dropping the partition should delete folder, so we should be able to recreate it again now
drop(partition);
assert!(keyspace.open_partition("default", Default::default).is_ok());
```

Comment (dharanad) [reply src/keyspace.rs:380]: My bad. I thought so, it should be in the else case must have interpreted the comments wrong. I will refactor

Comment (dharanad) [reply src/keyspace.rs]: I have moved this sanity check while createing a partition and added a test check to verify the behaviour. Thanks for your help

**Review by marvin-j97 — APPROVED** [review]
I changed the test case to check for the specific enum variant.

Good idea to move the marker check into the `Partition` itself. :+1:

**Comment by dharanad** [comment]
Thank you for you time @marvin-j97 . I would like to contribute more to this and other projects of yours. Please let me know how I can get involved.

**Comment by marvin-j97** [comment]
> Thank you for you time @marvin-j97 . I would like to contribute more to this and other projects of yours. Please let me know how I can get involved.

I'm currently pretty deep into finishing V2, so a lot of stuff is currently blocked or in progress.

There are two minor examples that are pretty low-hanging fruits - #48 and #49 - which I'm not really getting around to implementing right now.

**Pull request merged**